src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -5173,9 +5173,7 @@ void ClassFileParser::fill_instance_klass(InstanceKlass* ik,
   ik->set_has_nonstatic_concrete_methods(_has_nonstatic_concrete_methods);
   ik->set_declares_nonstatic_concrete_methods(_declares_nonstatic_concrete_methods);
 
-  if (_is_hidden) {
-    ik->set_is_hidden();
-  }
+  assert(!_is_hidden || ik->is_hidden(), "must be set already");
```

Comment:
Is this an optimization independent of the current change?

Reply:
No, it's needed by the current change because with these Flags.hpp implementations, we assert that they are only assigned once i.e. const after classfile parsing.  This was a second assignment.